### ISSUE TITLE
Eval always 'nan' when training on custom dataset

### ISSUE THREAD

**Issue by xiangsanliu** [open-mmlab/mmsegmentation#964]
here is the log:
```
+------------+-------+-------+
|   Class    |  IoU  |  Acc  |
+------------+-------+-------+
| background | 99.96 | 99.96 |
|  building  |  0.0  |  nan  |
+------------+-------+-------+
2021-10-14 18:03:45,347 - mmseg - INFO - Summary:
2021-10-14 18:03:45,348 - mmseg - INFO - 
+-------+-------+-------+
|  aAcc |  mIoU |  mAcc |
+-------+-------+-------+
| 99.96 | 49.98 | 99.96 |
+-------+-------+-------+
2021-10-14 18:03:45,350 - mmseg - INFO - Iter(val) [400]	aAcc: 0.9996, mIoU: 0.4998, mAcc: 0.9996, IoU.background: 0.9996, IoU.building: 0.0000, Acc.background: 0.9996, Acc.building: nan
2021-10-14 18:04:19,291 - mmseg - INFO - per class results:
2021-10-14 18:04:19,292 - mmseg - INFO - 
+------------+-------+-------+
|   Class    |  IoU  |  Acc  |
+------------+-------+-------+
| background | 99.93 | 99.93 |
|  building  |  0.0  |  nan  |
+------------+-------+-------+
2021-10-14 18:04:19,292 - mmseg - INFO - Summary:
2021-10-14 18:04:19,292 - mmseg - INFO - 
+-------+-------+-------+
|  aAcc |  mIoU |  mAcc |
+-------+-------+-------+
| 99.93 | 49.97 | 99.93 |
+-------+-------+-------+
2021-10-14 18:04:19,294 - mmseg - INFO - Iter(val) [400]	aAcc: 0.9993, mIoU: 0.4997, mAcc: 0.9993, IoU.background: 0.9993, IoU.building: 0.0000, Acc.background: 0.9993, Acc.building: nan
2021-10-14 18:04:53,328 - mmseg - INFO - per class results:
2021-10-14 18:04:53,328 - mmseg - INFO - 
+------------+-------+-------+
|   Class    |  IoU  |  Acc  |
+------------+-------+-------+
| background | 99.89 | 99.89 |
|  building  |  0.0  |  nan  |
+------------+-------+-------+
2021-10-14 18:04:53,328 - mmseg - INFO - Summary:
2021-10-14 18:04:53,329 - mmseg - INFO - 
+-------+-------+-------+
|  aAcc |  mIoU |  mAcc |
+-------+-------+-------+
| 99.89 | 49.95 | 99.89 |
+-------+-------+-------+
2021-10-14 18:04:53,330 - mmseg - INFO - Iter(val) [400]	aAcc: 0.9989, mIoU: 0.4995, mAcc: 0.9989, IoU.background: 0.9989, IoU.building: 0.0000, Acc.background: 0.9989, Acc.building: nan
```

**Comment by skaldesh** [comment]
By setting `reduce_zero_label=True`, you actually ignore the class with value 0. So in your dataset:  

> CLASSES = ("background", "building")
   PALETTE = [[0], [1]]

should probably be:

> CLASSES = ("building")
   PALETTE = [[1]]

**Comment by xiangsanliu** [comment]
> By setting `reduce_zero_label=True`, you actually ignore the class with value 0. So in your dataset:
> 
> > CLASSES = ("background", "building")
> > PALETTE = [[0], [1]]
> 
> should probably be:
> 
> > CLASSES = ("building")
> > PALETTE = [[1]]

Thanks, I'll try this.

**Comment by zijinY** [comment]
Have you solved such problem?

**Comment by RoyZhDec** [comment]
get the same problem, and I have already set reduce_zero_label=True. Anyone can help?

**Comment by hblab-haims** [comment]
Here my config.
![image](https://user-images.githubusercontent.com/89757661/145928832-4104cd2c-aec3-43a0-84df-7b2481d56ba3.png)

With this config, I can get model to predict my expect class `plate_segmentation` at very early iter, for example, 50 first iters. 
But after several hundred iters, model stop predicting `plate_segmentation`, which lead to acc = 0.
![image](https://user-images.githubusercontent.com/89757661/145929174-21d3e7a2-4af4-4626-8d84-a44d3f1bcf71.png)

Please give me some suggest to over this situation.

**Comment by RoyZhDec** [comment]
> Here my config. ![image](https://user-images.githubusercontent.com/89757661/145928832-4104cd2c-aec3-43a0-84df-7b2481d56ba3.png)
> 
> With this config, I can get model to predict my expect class `plate_segmentation` at very early iter, for example, 50 first iters. But after several hundred iters, model stop predicting `plate_segmentation`, which lead to acc = 0. ![image](https://user-images.githubusercontent.com/89757661/145929174-21d3e7a2-4af4-4626-8d84-a44d3f1bcf71.png)
> 
> Please give me some suggest to over this situation.


I also have the same problem. In the first few iterations, the model has a two-class classification ability. Later it converges better and better, and the loss drops normally. Instead, all models only output 1 label.

**Comment by maisonhai3** [comment]
> I also have the same problem. In the first few iterations, the model has a two-class classification ability. Later it converges better and better, and the loss drops normally. Instead, all models only output 1 label.

In my case, I found out that there is an error in my configuration. Look at the picture.
This is the mistake:
![image](https://user-images.githubusercontent.com/28621061/146249054-e7befd9f-fed2-4739-8035-bc9114495c4e.png)

Which should be:
![image](https://user-images.githubusercontent.com/28621061/146249355-a187fc4b-1bb4-463e-b956-d729ac04e0c1.png)

With `reduce_zero_label=True` in `train_pipeline`, it overwrites my custom dataset's `reduce_zero_label`:
![image](https://user-images.githubusercontent.com/28621061/146249791-e37f7d8c-6e43-4217-aae5-5a0293c04796.png)
At line 19th, `reduce_zero_label=False` is correct, but be overwritten by `train_pipeline`.

After removing `train_pipeline remove_zero_label`, it works.

Here my configs and results:

![image](https://user-images.githubusercontent.com/28621061/146250404-c3015ee7-e16f-4701-bea6-0d10c0fdb3ce.png)

**Comment by maisonhai3** [comment]
> Dataset config:


> My Custom dataset

I think they are fine. Maybe problems come from elsewhere. 

This [advice on Loss and learning rate](https://www.kaggle.com/bigironsphere/loss-function-library-keras-pytorch#:~:text=Usage-,tips,-chevron_right) from kaggle could be helpful.

**Comment by ke-dev** [comment]
Hi, I have the same problem, has your problem been solved?

**Comment by KongCang** [comment]
same config, some data work well, some data get nan! is the lr need to be smaller?

**Comment by MengzhangLI** [comment]
> 

Perhaps it is because some data has many foreground labels while some image only has background and certain foreground which is ignored in calculating loss (denominator is zero) thus it got NaN.

**Comment by zhang-bite** [comment]
i also met this problem, i finally find out the reason is that my 'SegmentationClass' is [0, 255], but mmsegmentation needs [0, 1], so i change that and my problem be sovled.

**Comment by akash-3196** [comment]
> i also met this problem, i finally find out the reason is that my 'SegmentationClass' is [0, 255], but mmsegmentation needs [0, 1], so i change that and my problem be sovled.

Hi, can you explain a little bit what exactly did you change in the labels to convert from [0,255] to [0,1]?
